maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/DistroGenerator.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+import org.apache.commons.io.IOUtils;
+
+public class DistroGenerator {
+
+	public static String generateDistroProperties(String distroTemplateUrl, String distroPomUrl, String spaConfigUrl, String spaPomUrl) {
+		try {
+			Document distroPom = parseXMLFromURL(distroPomUrl);
+			Document spaModulePom = parseXMLFromURL(spaPomUrl);
+			Properties backendProperties = getPomDocument((Element) distroPom.getElementsByTagName("properties").item(0));
+			Properties configuarationProperties = getPomDocument((Element) distroPom.getElementsByTagName("parent").item(0));
+			String spaModuleVersion = spaModulePom.getElementsByTagName("version").item(0).getTextContent();
+			return getModuleEntries(distroTemplateUrl, backendProperties) +
+					getFrontendEntriesFromSpaConfig(spaConfigUrl) +
+					getConfigurationEntry(configuarationProperties) + getSpaModuleEntry(spaModuleVersion);
+		}
+		catch (ParserConfigurationException | IOException | SAXException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private static String getSpaModuleEntry(String spaModuleVersion) {
+		return "omod.spa=" + spaModuleVersion;
+	}
+
+	private static String getConfigurationEntry(Properties properties) {
+		return "config." + properties.getProperty("artifactId") + "-config=" + properties.getProperty("version") + "\n";
+	}
+
+	public static String getFrontendEntriesFromSpaConfig(String url) {
+		try {
+			HttpClient httpClient = HttpClientBuilder.create().build();
+			HttpGet request = new HttpGet(url);
+			HttpResponse response = httpClient.execute(request);
+			HttpEntity entity = response.getEntity();
+			InputStream inputStream = entity.getContent();
+			String jsonData = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+			inputStream.close();
+			ObjectMapper objectMapper = new ObjectMapper();
+			JsonNode jsonNode = objectMapper.readTree(jsonData);
+			JsonNode frontendModules = jsonNode.get("frontendModules");
+			StringBuilder formattedModules = new StringBuilder();
+			Iterator<Map.Entry<String, JsonNode>> modulesIterator = frontendModules.fields();
+			while (modulesIterator.hasNext()) {
+				Map.Entry<String, JsonNode> moduleEntry = modulesIterator.next();
+				String moduleName = moduleEntry.getKey();
+				String moduleVersion = moduleEntry.getValue().asText();
+				String formattedModule = "spa.frontendModules." + moduleName + "=" + moduleVersion;
+				formattedModules.append(formattedModule).append("\n");
+			}
+			return formattedModules.toString();
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+		return "";
+	}
+
+	public static String getModuleEntries(String url, Properties properties) {
+		try {
+			String distroTemplate = fetchFileContent(url);
+			if (distroTemplate != null)
```

Comment:
https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-CodeStyle

Reply:
I fixed this formatting issue.